src/helpers/hooks.tsx

```diff
@@ -17,3 +17,35 @@ export function useConfirmOnBrowserUnload(hasUnsavedChanges: boolean = false) {
     return () => window.removeEventListener('beforeunload', callback);
   });
 }
+
+/**
+ * Debounce calls to make sure they do not execute too frequently such as when
+ * a user is typing and a call needs to be made to the API
+ * @param value Value to be debounced
+ * @param delay Time in ms to wait for since the last call
+ */
+export function useDebounce(value: string, delay: number = 1000) {
```

Comment:
Is this being used anywhere?

Reply:
Yes:

1. `src/containers/pages/InterventionPlan/PlanDefinitionList/index.tsx
2. `src/containers/pages/IRS/plans/index.tsx`